src/ui/public/chrome/directives/kbn_chrome.js

```diff
@@ -43,6 +43,29 @@ export default function (chrome, internals) {
         // and some local values
         chrome.httpActive = $http.pendingRequests;
         $scope.notifList = require('ui/notify')._notifs;
+
+        // App switcher functionality.
+        function updateAppSwitcher(isOpen) {
+          $scope.appSwitcherButton = {
+            classes: isOpen ? 'app-switcher-link--close' : undefined,
+            title: isOpen ? 'Collapse' : 'Expand',
+            tooltip: isOpen ? 'Collapse side bar' : 'Expand side bar',
+          };
+        }
+
+        $scope.isAppSwitcherOpen = appSwitcherState.isOpen();
+        updateAppSwitcher($scope.isAppSwitcherOpen);
+
+        $scope.toggleAppSwitcher = event => {
+          event.preventDefault();
+          $scope.isAppSwitcherOpen = !$scope.isAppSwitcherOpen;
+          appSwitcherState.setOpen($scope.isAppSwitcherOpen);
+          updateAppSwitcher($scope.isAppSwitcherOpen);
+
+          // Notify visualizations, e.g. the dashboard, that they should re-render.
+          $scope.$root.$broadcast('ready:vis');
```

Comment:
Is this how the visualizations get re-rendered on window re-size? I only see this getting emitted from visualize.js and doc_table.js atm, so I'm a little worried this might be misusing an existing event, but I don't know enough about Visualize to be sure.